smartsim/settings/launchSettings.py

```diff
@@ -67,7 +67,7 @@ def launcher(self) -> str:
         return self._launcher.value
 
     @property
-    def launch_args(self) -> LaunchArgBuilder:
+    def launch_args(self) -> LaunchArgBuilder[t.Any]:
```

Comment:
> maybe it is worth being consisent in the code base and just using t.Union until we change everything over?

I was actually hoping that, since the core refactor is going to be touching basically every file in SmartSim anyway, we could start moving to the new syntax now in a "boy-scouts-leave-it-better-than-how-you-found-it" sort of soft rule way (especially since 3.9 only has about one year left of support).

More than willing to be talked out of this position though! If there is a strong instance, to keep the "old-style" syntax I'd be more than willing to revert the changes!

Reply:
> Also line 65 the comment might be better as "Return the launcher type"

Updated!

Reply:
Decided to make this change over piece-meal, a ticket will be added to remove any `t.Union`, `t.Optional`, `t.List`, etc. when the `smartsim_refactor` branch goes into develop